devdoc/userdoc/AvoidlegacyFlag.md

```diff
@@ -0,0 +1,45 @@
+# Why You Should Avoid Using the -legacy Compiler Option
+
+## Overview of the -legacy Flag
+
+The **-legacy** flag was provided as a short-term solution to give you a chance to modify your code for the new semantics. It was always intended to be a temporary solution -- in fact, it was marked as deprecated when it was introduced.
+
+The updated **IMPORT** rules are more robust and predictable. They prevent namespace collisions and align with current best practices. The updated **WHEN** behavior provides deterministic scoping, order independence, and improved optimization.
+
+Aligning with current standards helps teams collaborate more effectively and reduces onboarding friction for new developers.
+
+## Reasoning for Avoiding the -legacy Flag
+
+The legacy flag creates several issues:
+
+1. **Symbol Conflicts**: Can create symbol conflicts in your code when you upgrade the platform.
+2. **Syntax Check Failures**: Can cause valid syntax to fail syntax checks.
+3. **Deprecated Behaviors**: Enables deprecated behaviors that will be removed, risking future code issues.
+4. **Performance Impact**: Can disable newer optimizer paths and reduce performance. Compiles may take longer if unrelated code matches by name.
```

Comment:
The sentence is wrong/adds nothing to the second.

Reply:
revised